pkg/pattern/drain/drain_test.go

```diff
@@ -1 +1,147 @@
 package drain
+
+import (
+	"bufio"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var drainConfig = &Config{
+	// At training, if at the depth of LogClusterDepth there is a cluster with
+	// similarity coefficient greater that SimTh, then the log message is added
+	// to that cluster. Otherwise, a new cluster is created.
+	//
+	// LogClusterDepth should be equal to the number of constant tokens from
+	// the beginning of the message that likely determine the message contents.
+	//
+	//  > In this step, Drain traverses from a 1-st layer node, which
+	//  > is searched in step 2, to a leaf node. This step is based on
+	//  > the assumption that tokens in the beginning positions of a log
+	//  > message are more likely to be constants. Specifically, Drain
+	//  > selects the next internal node by the tokens in the beginning
+	//  > positions of the log message
+	LogClusterDepth: 8,
+	// SimTh is basically a ratio of matching/total in the cluster.
+	// Cluster tokens: "foo <*> bar fred"
+	//       Log line: "foo bar baz qux"
+	//                  *   *   *   x
+	// Similarity of these sequences is 0.75 (the distance)
+	// Both SimTh and MaxClusterDepth impact branching factor: the greater
+	// MaxClusterDepth and SimTh, the less the chance that there will be
+	// "similar" clusters, but the greater the footprint.
```

Comment:
Super cool thank you for that !

Reply:
I can't take credit for this! I copied the descriptions and parameters over from https://github.com/grafana/loki/blob/491d2513e7469809d5b6962359305b9f7e6cb250/pkg/pattern/stream.go#L30 in order to get started. 
But now I think it makes more sense to unify it under the DefaultConfig function inside the drain package, at least until it becomes clear that we'd need different configs for different sets of logs.